### DISCUSSION TITLE
OpenBSD Build

### DISCUSSION THREAD

**Discussion by ghost** [red/red#5107] [Q&A]
I am looking to use Red on OpenBSD. Unfortunately I was unable to find build instructions, only a mention that one needs REBOL. REBOL in turn does not seem to be available for OpenBSD. Any help would be appreciated.

Ideally, the Red project publishes a build for OpenBSD (amd64).

**Comment by hiiamboris** [comment]
Building instructions listed at https://github.com/red/red/
Rebol 2 can be downloaded here: http://www.rebol.com/download-core.html

Both R2 and Red are 32-bit at the moment, so 32-bit libraries are needed. Instructions for FreeBSD can be found here: https://www.red-lang.org/p/download.html 
If these don't work for OpenBSD, once you figure it out feel free to list OpenBSD-specific process here so we can add it to that page.

**Reply by ghost** [reply]
Argh, so that's another hurdle - OpenBSD's amd64 build does not run 32-bit binaries. So I guess the question is, when Red will have a 64-bit support and can be built without R2 (which seems to be stuck in 32-bit land). 😢

**Reply by hiiamboris** [reply]
In that case you're out of luck to run it natively for now and there's no ETA on 64-bit builds yet, not to mention for self-hosted Red. Mac people are facing the same issue.
Meanwhile, you can take a look at https://github.com/red/red/wiki/[LINKS]-Unofficial-Red-images page to see if docker is an acceptable solution.
